src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala

```diff
@@ -264,38 +265,39 @@ object TezosTypes {
 
     case class Reveal(
         kind: String,
-        source: ContractId,
+        source: PublicKeyHash,
         nonce: Int,
         public_key: PublicKey,
         result: OperationResult.Reveal
     ) extends InternalOperationResult
 
     case class Transaction(
         kind: String,
-        source: ContractId,
+        source: PublicKeyHash,
         nonce: Int,
         amount: PositiveBigNumber,
         destination: ContractId,
         parameters: Option[Micheline],
         result: OperationResult.Transaction
     ) extends InternalOperationResult
 
+    /* some fields are only kept for backward-compatibility, as noted*/
     case class Origination(
         kind: String,
-        source: ContractId,
+        source: PublicKeyHash,
         nonce: Int,
-        manager_pubkey: PublicKeyHash,
+        manager_pubkey: Option[PublicKeyHash], // retro-compat from protocol 5+
```

Comment:
Can you explain why we need to maintain retro compatibility?

Reply:
I now understand you were trying to keep Lorre consistent with data from nodes running older protocols. Maybe this is still possible. I will investigate.